### ISSUE TITLE
Instructor example is broken

### ISSUE THREAD

**Issue by olafgeibig** [MadcowD/ell#304]
When running instructor_ex.py from the ell repo it fails.

- ell 0.0.14
- instrucutor 1.5.2

```
Traceback (most recent call last):
  File "/Users/GEO5BE4/work/ai-experiments/ell-test/ell_test/instructor_ex.py", line 54, in <module>
    print(extract_user("Jason is 25 years old"))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/GEO5BE4/work/ai-experiments/ell-test/.venv/lib/python3.12/site-packages/ell/lmp/_track.py", line 64, in tracked_func
    return func_to_track(*fn_args, **fn_kwargs, _invocation_origin=invocation_id)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/GEO5BE4/work/ai-experiments/ell-test/.venv/lib/python3.12/site-packages/ell/lmp/complex.py", line 68, in model_call
    (result, final_api_params, metadata) = provider.call(ell_call, origin_id=_invocation_origin, logger=_logger if should_log else None)
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/GEO5BE4/work/ai-experiments/ell-test/.venv/lib/python3.12/site-packages/ell/provider.py", line 125, in call
    provider_resp = call(**final_api_call_params)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/GEO5BE4/work/ai-experiments/ell-test/.venv/lib/python3.12/site-packages/instructor/client.py", line 119, in create
    return self.create_fn(
           ^^^^^^^^^^^^^^^
  File "/Users/GEO5BE4/work/ai-experiments/ell-test/.venv/lib/python3.12/site-packages/instructor/patch.py", line 279, in new_create_sync
    response_model, new_kwargs = handle_response_model(
                                 ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/GEO5BE4/work/ai-experiments/ell-test/.venv/lib/python3.12/site-packages/instructor/process_response.py", line 668, in handle_response_model
    new_kwargs["messages"] = convert_messages(new_kwargs["messages"], mode)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/GEO5BE4/work/ai-experiments/ell-test/.venv/lib/python3.12/site-packages/instructor/multimodal.py", line 115, in convert_messages
    converted_content = convert_contents(content, mode)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/GEO5BE4/work/ai-experiments/ell-test/.venv/lib/python3.12/site-packages/instructor/multimodal.py", line 99, in convert_contents
    raise ValueError(f"Unsupported content type: {type(content)}")
ValueError: Unsupported content type: <class 'dict'>
```

**Comment by reecelikesramen** [comment]
Hey @olafgeibig I was able to recreate your issue. This actually appears to be an issue with the Instructor package.

Instructor expects to handle individual *text* messages from like this: `{"role": <role>, "content": <text>}`, but when ell calls the provider it gives it this: `{"role": <role>, "content": [{"type": "text", "content": <text>}]}`. This is perfectly valid syntax if you wanted to pass multiple content blocks in a single message.

I have verified that Instructor will raise an error when given messages like this by modifying an example from that repo. No changes need to be made by Ell. I will open a pull request on the Instructor repo that should fix this.

This is a diff correcting the issue in Instructor code, with this patch in place the example works as expected:
```
--- /Users/rholmdahl/projects/ell/.venv/lib/python3.12/site-packages/instructor/multimodal.py	2024-10-13 18:26:28
+++ /Users/rholmdahl/projects/ell/multimodal_modified.py	2024-10-13 18:26:11
@@ -95,6 +95,11 @@
                 raise NotImplementedError("Gemini is not supported yet")
             else:
                 converted_contents.append(content.to_openai())
+        elif isinstance(content, dict):
+            # could also check:
+            # content.get("type") == "text" and isinstance(content.get("text"), str) 
+            # this works for now but is less safe?
+            converted_contents.append(content)
         else:
             raise ValueError(f"Unsupported content type: {type(content)}")
     return converted_contents
```

@MadcowD you can close this issue

**Comment by reecelikesramen** [comment]
Looks like my fix was already implemented into Instructor in [this PR](https://github.com/instructor-ai/instructor/commit/4d138e2dcda48eb786bddf0b32fd2d94fce78a7c) but was not included for latest release 1.5.2. Probably will released in a few days and this won't be an issue anymore.

**Comment by olafgeibig** [comment]
Closing issue, awaiting fix in new instructor release.
